File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_CSharp/objetosgui/TabelaItemComparer.cs
/*
 * Created by SharpDevelop.
 * User: henrique.souza
 * Date: 09/07/2015
 * Time: 17:06
 * 
 * To change this template use Tools | Options | Coding | Edit Standard Headers.
 */
using System;
using System.Collections;
using System.Windows.Forms;

namespace HFSGuardaDiretorio.objetosgui
{
	/// <summary>
	/// Description of TabelaItemComparer.
	/// </summary>
	public class TabelaItemComparer: IComparer {
	    private int col;
	    private SortOrder ordem;

	    public TabelaItemComparer() {
	        col=0;
	        ordem = SortOrder.Ascending;
	    }

	    public TabelaItemComparer(int coluna, SortOrder ordem)
	    {
	        col=coluna;
	        this.ordem = ordem;
	    }

	    public int Compare(object x, object y)
	    {
	        int retornoVal = -1;
	        retornoVal = String.Compare(((ListViewItem)x).SubItems[col].Text,
	        	((ListViewItem)y).SubItems[col].Text);
			
	        if (ordem == SortOrder.Ascending) {
	            retornoVal *= -1;        
	        }
	        
	        return retornoVal;
	    }
	}
}

File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_VBNet/objetosgui/TabelaItemComparer.vb
'
' * Created by SharpDevelop.
' * User: henrique.souza
' * Date: 09/07/2015
' * Time: 17:06
' * 
' * To change this template use Tools | Options | Coding | Edit Standard Headers.
' 

Imports System.Collections
Imports System.Windows.Forms

Namespace objetosgui
	''' <summary>
	''' Description of TabelaItemComparer.
	''' </summary>
	Public Class TabelaItemComparer
		Implements IComparer
		Private col As Integer
		Private ordem As SortOrder

		Public Sub New()
			col = 0
			ordem = SortOrder.Ascending
		End Sub

		Public Sub New(coluna As Integer, ordem As SortOrder)
			col = coluna
			Me.ordem = ordem
		End Sub

		Public Function Compare(x As Object, y As Object) As Integer Implements IComparer.Compare
			Dim retornoVal As Integer = -1
			retornoVal = [String].Compare(DirectCast(x, ListViewItem).SubItems(col).Text, DirectCast(y, ListViewItem).SubItems(col).Text)

			If ordem = SortOrder.Ascending Then
				retornoVal *= -1
			End If

			Return retornoVal
		End Function
	End Class
End Namespace
